js/index.js

```diff
@@ -1,39 +1,184 @@
+                data.reverse();
             }
+            else if (isNaN(criteria)) {
+
+                if (criteria == 'place') {
+                    //special rule for strings
+                    data.sort((a, b) => _treatString(a, b, criteria));
+                }
+                else {
+                    // sort by properties
+                    data.sort((a, b) => b.properties[criteria] - a.properties[criteria]);
+                }
+            }
+            else {
+                //sort by geometry
+                data.sort(function(a, b){
+                    return b.geometry.coordinates[criteria] - a.geometry.coordinates[criteria];
+                }); 
+            }
+
+            return data;
+        };
+
+
+    return {
+        updateTable : function(criteria, data, order) {
+            data = _sortBy(criteria.getAttribute('value'), data, order);
+            _clearTable();
+            _fillTable(this.assembleTable(data));
+
+        },
+        assembleTable : function(data) {
+
+            //assembles table rows
+            let table = '';
+            
+            data.forEach(element => {
+                table += `
+                    <tr>
+                        <td>${element.properties.place}</td>
+                        <td>${_getTime(element.properties.time)}</td>
+                        <td><span style='color:${element.properties.alert}'>&#9654</span></td>
+                        <td>${element.geometry.coordinates[0]}</td>
+                        <td>${element.geometry.coordinates[1]}</td>
+                        <td>${element.geometry.coordinates[2]} Km</td>
+                    </tr>
+                `
+            });
+
+            return table;
+        }
+    };
+
+})();
+
+const UIController = (function() {
+
+    const _DOMString = {
+        summary : '.summary span.amount',
+        table : 'tablePrint',
+        tableHead : '.table th'
+    }
+
+    // Jquery Animation
+    const _countTotalAnimation = () => {
+        $(_DOMString.summary).each(function () {
+            $(this).prop('Counter',0).animate({
+                Counter: $(this).text()
+            }, {
+                duration: 4000,
+                easing: 'swing',
+                step: function (now) {
+                    $(this).text(Math.ceil(now));
+                }
+            });
         });
+
+    }
+
+    return {
+        DOM : _DOMString,
+
+        totalEarthquakes : amount => {
+            document.querySelector(_DOMString.summary)
+            .textContent = amount;
+            _countTotalAnimation()
+        },
+
+        PrintTable : table => {
+            document.getElementById(_DOMString.table)
+            .innerHTML = table;
+        }
     }
-};
 
-xmlhttp.open('GET', 'https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/significant_month.geojson', true);
-xmlhttp.send();
+})();
+
+
+const main = (function(UICtrl, utils) {
+
+    // This will be used to info the current col to be sorted
```

Comment:
This will be used to determine col to be sorted. (Ajuste o comentário por favor).